### PR TITLE
feat: card api integration for update/edit

### PR DESCRIPTION
**Pull request by bharathraj25**
* added complete CRUD operation api integration
* refactored some file casing
* resolved some previous PR comments

### CODE REVIEW

File: src/components/cards/CustomCard.js [hunk 1]
```diff
@@ -37,7 +39,17 @@ const CardHeader = styled(Box)(() => ({
 
 export default function CustomCard({ checkBox, data, type }) {
   const [checked, setChecked] = useState(data.status === STATUS.completed);
-  const { setError, setSuccess } = useContext(ErrorContext);
+  const { setError, setSuccess } = React.useContext(ErrorContext);
+
```
Comment (sugumar-mani) [inline src/components/cards/CustomCard.js:43]: unnecessary empty line

File: src/components/cards/CustomCard.js [hunk 2]
```diff
@@ -91,41 +110,51 @@ export default function CustomCard({ checkBox, data, type }) {
...
-            <Typography variant="caption" display="block" gutterBottom>
-              {dateGetter(data.createdAt, "createdAt")}
-            </Typography>
-          </CardHeader>
-          <Box>
-            <Tooltip title={data.note}>
+        <CardActionArea onClick={handleClickOpen}>
+          <CardContent >
+            <CardHeader>
+              {
```
Comment (sugumar-mani) [inline src/components/cards/CustomCard.js:118]: can be a single line

File: src/components/poNotesComponents/AddPONotes.js [hunk 3]
```diff
@@ -1,215 +1,23 @@
...
-function Loading() {
-  return <Box>Loading...</Box>;
-}
-Item.propTypes = {
-  entity: Proptypes.shape({
-    name: Proptypes.string.isRequired,
-    char: Proptypes.string.isRequired,
-  }).isRequired,
-};
+export default function AddPoNotes() {
```
Comment (sugumar-mani) [inline src/components/poNotesComponents/AddPONotes.js:6]: AddPoNotes to `AddPONotes`

File: src/components/cards/CustomCard.js [hunk 4]
```diff
@@ -91,41 +110,51 @@ export default function CustomCard({ checkBox, data, type }) {
...
+            <Box sx={{ position: 'relative', bottom: 0, top: 35}}>
+              {renderdueDate()}
+              <Box sx={{display:'flex',justifyContent:'space-between'}}>
+              <Stack direction="row" spacing={-1} mb={4} pl={1}>
+                {
+                  collaborators.map((names) => <Avatar {...stringAvatar(names)} />)
+                }
+              </Stack>
+              <Box pr={2}>
+            {renderLink()}
```
Comment (sugumar-mani) [inline src/components/cards/CustomCard.js:152]: alignment issue

**Review by sugumar-mani — APPROVED** [review]
please fix the comments and merge

File: src/components/poNotesComponents/AddPONotes.js (hunk 3 above)
Comment (samim-hossain-mondal) [reply src/components/poNotesComponents/AddPONotes.js:6]: fixed it

File: src/components/cards/CustomCard.js (hunk 4 above)
Comment (samim-hossain-mondal) [reply src/components/cards/CustomCard.js:152]: fixed the aligment

File: src/components/cards/CustomCard.js (hunk 1 above)
Comment (samim-hossain-mondal) [reply src/components/cards/CustomCard.js:43]: removed empty line

File: src/components/cards/CustomCard.js (hunk 2 above)
Comment (samim-hossain-mondal) [reply src/components/cards/CustomCard.js:118]: fixed it

**Pull request merged**